src/flows/shared/BucketSelector/CreateBucketDropdownItem.tsx

```diff
@@ -0,0 +1,52 @@
+// Libraries
+import React, {FC, useEffect} from 'react'
+import {useDispatch, useSelector} from 'react-redux'
+
+// Components
+import {Dropdown} from '@influxdata/clockface'
+
+// Actions
+import {checkBucketLimits, LimitStatus} from 'src/cloud/actions/limits'
+import {showOverlay, dismissOverlay} from 'src/overlays/actions/overlays'
+
+// Utils
+import {getBucketLimitStatus} from 'src/cloud/utils/limits'
+
+// Types
+import {Bucket} from 'src/types'
+
+// Constants
+import {CLOUD} from 'src/shared/constants'
+
+interface Props {
+  onCreateBucket: (bucket: Bucket) => void
+}
+
+const CreateBucketDropdownItem: FC<Props> = ({onCreateBucket}) => {
+  const limitStatus = useSelector(getBucketLimitStatus)
+  const dispatch = useDispatch()
+  useEffect(() => {
+    // Check bucket limits when component mounts
+    dispatch(checkBucketLimits())
```

Comment:
is there some kind of caching behind this? You should totally be able to call it like this in theory, but it doesn't feel right to lean on the convention that this will only render once. it's always one dom level change from exploding a bunch of network requests, like a bomb waiting for some misplaced input field to set it off

Reply:
I didn't write it, but I think it's some insurance that bucket limits exist before trying to render this component

Reply:
Open to suggestions here